### PR TITLE
HW11 Task01

### PR DESCRIPTION
**Pull request by LinaKrevetochka**

### CODE REVIEW

File: hw/hw_11_task_01.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+class SimplifiedEnum(type):
+    def __new__(cls, name, bases, dct):  # noqa
+        cls_instance = super().__new__(cls, name, bases, dct)
+        cls_instance.__values__ = dct[f"_{name}__keys"]
+        for key in cls_instance.__values__:
+            setattr(cls_instance, key, key)
+        return cls_instance
+
+    def __iter__(cls):
```
Comment (Cjkjvfnby) [inline hw/hw_11_task_01.py]: Is it required?

File: hw/hw_11_task_01.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        return (name for name in cls.__values__)
+
+    def __len__(cls):
```
Comment (Cjkjvfnby) [inline hw/hw_11_task_01.py]: Is it required?

File: hw/hw_11_task_01.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    __keys = ("XL", "L", "M", "S", "XS")
+assert ColorsEnum.RED == "RED"
+assert SizesEnum.XL == "XL"
+"""
+
+
+class SimplifiedEnum(type):
+    def __new__(cls, name, bases, dct):  # noqa
+        cls_instance = super().__new__(cls, name, bases, dct)
+        cls_instance.__values__ = dct[f"_{name}__keys"]
```
Comment (Cjkjvfnby) [inline hw/hw_11_task_01.py]: `__values__ ` looks like dandr method but it is not in Python specification.

File: hw/hw_11_task_01.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        for key in cls_instance.__values__:
+            setattr(cls_instance, key, key)
```
Comment (Cjkjvfnby) [inline hw/hw_11_task_01.py]: You can modify `dct` before class creation instead of adding attributes after.

File: test/test_01_hw_11.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from hw.hw_11_task_01 import SimplifiedEnum
+
+
+class ColorsEnum(metaclass=SimplifiedEnum):
+    __keys = ("RED", "BLUE", "ORANGE", "BLACK")
+
+
+def test_enum():
+    assert ColorsEnum.RED == "RED"
```
Comment (Cjkjvfnby) [inline test/test_01_hw_11.py:9]: Why did you choose red only?

_I see a red door
And I want it painted black
No colors anymore
I want them to turn black._

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw/hw_11_task_01.py (hunk 1 above)
Comment (LinaKrevetochka) [reply hw/hw_11_task_01.py]: I thought we must create a class that acts like real enum.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**